core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorConcurrencyTest.scala

```diff
@@ -278,37 +302,48 @@ class GroupCoordinatorConcurrencyTest extends AbstractCoordinatorConcurrencyTest
     }
   }
 
-  class CommitTxnOffsetsOperation extends CommitOffsetsOperation {
+  /**
+   * @param needLock true to make CompleteTxnOperation happen after CommitTxnOffsetsOperation. It presents error when
+   *                 both CommitTxnOffsetsOperation and CompleteTxnOperation are executed on parallel.
```

Comment:
Perhaps change the comment to sth like the following?

"Setting to true to make CompleteTxnOperation and CommitTxnOffsetsOperation complete atomically since they don't typically overlap. Otherwise CompleteTxnOperation may see a pending offsetAndMetadata without an appendedBatchOffset."